src/automate.py

```diff
@@ -124,19 +124,21 @@ def run_executable(executable, arguments, plat, driver=None, top_level=True) :
     performance_setup = []
     performance_main = []
     performance_tear_down = []
+
     if "setup" in executable :
-        driver, performance_setup = run_executable(executable["setup"], arguments, plat, driver, False)
+        driver, setup_performance_data = run_executable(executable["setup"], arguments, plat, driver, False)
+        performance_setup = setup_performance_data['details']['setup']
 
     set_platform(plat)
 
     input_file, program = parse_executable(executable)
-
     locked_program, driver, performance_main = run_program(program, platform, arguments, executable["name"], 1, driver)
 
     saved_locked_program(input_file, locked_program)
 
     if "tear-down" in executable :
-        driver, performance_tear_down = run_executable(executable["tear-down"], arguments, plat, driver, False)
+        driver, tear_down_performance_data = run_executable(executable["tear-down"], arguments, plat, driver, False)
+        performance_tear_down = setup_performance_data['details']['tear-down']
```

Comment:
I think I have used double quotes everywhere. The need for a linter is pretty obvious. I will set it up. But till then pls stick to the conventions used everywhere in the project. Find `'` replace `"`